### ISSUE TITLE
Allow training "data" configuration variable to specify an array of models to train

### ISSUE THREAD

**Issue by vibhutibindal** [RasaHQ/rasa_nlu#467]
**rasa NLU version** (e.g. `0.8.0`):

**Used backend / pipeline** (`spacy_sklearn`, ...):

**Operating system** (windows, osx, ...):

**Issue**:
i am trying to built multiple chatbots and i need  sample json file for each chatbot.so in my code i have to call train method for each chatbot.

configuration file:
{
  "pipeline": "spacy_sklearn",
  "path": "./models",
  "data": ["./data/sample1.json","./data/internal.json"]
}
`
config=RasaNLUConfig("config.json")
for i in (0,1):

 training_data = load_data(config.data[i])
 trainer = Trainer(RasaNLUConfig("config.json"))
 trainer.train(training_data)
 model_directory = trainer.persist('./models/')
`
can anyone suggets me how to optimize it if the no of file become too large or is there any rasa nlu prebuilt method to train all the models at the same time.

```

**Comment by wrathagom** [comment]
To my knowledge Rasa itself doesn't provide any tools or methods for doing this. Can I ask why you're needing to train such a large number of models at the same time? I would think it would be easier to maintain if you trained each one as it changed rather than all at once.

We're using docker and spin up a new docker container for training and close it down when it is completed. This allows us to parallelize as many training instances as our hardware can handle.

@tmbo @PHLF any comments on the `python` way of doing it.

**Comment by vibhutibindal** [comment]
i agree with you on the part that it is easier to update the model as it is changed but i am having a use case during a boot up process where i want all the models to need to be updated as soon as the new data comes in.This will be basically a salvage mechanism which will be triggered during the boot up for loading the fresh data in case my job scheduler fails and i don't know which models are up to date.This will be my cold start strategy to start with all my model updated.

**Comment by wrathagom** [comment]
Makes sense, I'll rename and re-lable this, but it will likely just go to a help wanted label unless @tmbo or someone on the Rasa team have the time to do it.
